CRM/Contribute/Selector/Search.php

```diff
@@ -402,6 +402,13 @@ public function &getRows($action, $offset, $rowCount, $sort, $output = NULL) {
         }
       }
 
+      $row['balance_due'] = CRM_Core_BAO_FinancialTrxn::getPartialPaymentWithType(
+        $result->contribution_id,
```

Comment:
@eileenmcnaughton @colemanw and I just looked through this function and it does two calls to the db which don't look bad. This .tpl is used on contributions tab and also on search results for Search: Find Contribution. But in the latter it's only going to be perhaps 100 or so records. So we're not too concerned after code review.